reference-implementation/lib/writable-stream.js

```diff
@@ -932,26 +947,28 @@ function WritableStreamDefaultControllerGetBackpressure(controller) {
 function WritableStreamDefaultControllerError(controller, e) {
   const stream = controller._controlledWritableStream;
 
-  assert(stream._state === 'writable' || stream._state === 'closing');
-
-  const oldState = stream._state;
+  assert(stream._state === 'writable');
 
   stream._state = 'errored';
   stream._storedError = e;
 
-  if (stream._pendingAbortRequest === undefined && stream._writer !== undefined) {
-    let readyPromiseIsPending = false;
-    if (oldState === 'writable' &&
-        WritableStreamDefaultControllerGetBackpressure(stream._writableStreamController) === true) {
-      readyPromiseIsPending = true;
+  // TODO(tyoshino): Given that writer.abort() fails immediately when the stream has been errored, shouldn't
```

Comment:
I'm not sure I completely understand this issue; a quick code snippet or test case illustrating the two alternative sunder discussion would be good.

I'm going to assume the question is, given `const p = var writer.abort(); controller.error();`, should `p` reject immediately, or should it follow the underlying sink's abort() return value?

If my understanding is correct, then my opinion is: if an underlying sink abort is underway already, it seems reasonable to let that run to completion instead of rejecting the writer.abort() promise.